Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp

```diff
@@ -32,21 +33,42 @@ bool isDetectorFixedInBank(const ComponentInfo &compInfo,
   return false;
 }
 
-bool isAnyBank(const ComponentInfo &compInfo, const size_t &idx) {
-  // if component at index is not a detector
+/*
+* Function: isSaveableBank. Returns true if the index in the Instrument cache is
+* a detector bank, ignoring tubes. otherwise returns false. Used by
+* SaveInstrument to find and save NXdetectors from memory to file.
+*
+* @param compInfo : Geometry::ComponentInfo Instrument cache containing the
+* component info.
+* @param detInfo :  Geometry::DetectorInfo Instrument cache containing the
+* detector info.
+* @param idx : size_t index of component
+* @return true if component at index is bank, false otherwise.
+
+*/
+bool isSaveableBank(const ComponentInfo &compInfo, const DetectorInfo &detInfo,
+                    const size_t idx) {
+  // return false if is a detector.
   if (compInfo.isDetector(idx))
     return false;
+  // needs to ignore if index is the sample
+  if (compInfo.sample() == idx)
+    return false;
+  // needs to ignore if index is the source
+  if (compInfo.source() == idx)
+    return false;
+  // a bank must have a parent, skip this block if the component does not.
   if (compInfo.hasParent(idx)) {
     size_t parent = compInfo.parent(idx);
     auto parentType = compInfo.componentType(parent);
     auto childType = compInfo.componentType(idx);
-    if (compInfo.detectorsInSubtree(idx).size() != 0) {
-      // if parent is not a detector bank
+    if (detInfo.size() != 0) {
```

Comment:
I'm not sure this check is actually write - my mistake. I think we can actually remove this part of the check entirely and avoid passing in the `detectorInfo`

Reply:
done